src/app/clients/family-members/add-family-members/add-family-members.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { FormGroup, FormBuilder, Validators, FormArray, FormControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { NgForm } from '@angular/forms';
+import { Router } from '@angular/router';
+import { ClientsService } from '../../clients.service';
+
+@Component({
+  selector: 'mifosx-add-family-members',
+  templateUrl: './add-family-members.component.html',
+  styleUrls: ['./add-family-members.component.scss']
+})
+export class AddFamilyMemberComponent implements OnInit {
+
+  /** add family member entry form. */
+  familyMemberForm: FormGroup;
+  genderIdData: any;
+  maritalStatusIdData: any;
+  professionIdData: any;
+  relationshipIdData: any;
+  paramsSubscription: Subscription;
+  id: number;
+
+
+
+  constructor(private formBuilder: FormBuilder,
+    private clientService: ClientsService,
+    private route: ActivatedRoute,
+    private router: Router) {
+    this.route.data.subscribe((data: { familyMember: any }) => {
+      this.genderIdData = data.familyMember.genderIdOptions;
+      this.maritalStatusIdData = data.familyMember.maritalStatusIdOptions;
+      this.professionIdData = data.familyMember.professionIdOptions;
+      this.relationshipIdData = data.familyMember.relationshipIdOptions;
+    });
+  }
+
+  ngOnInit() {
+    this.paramsSubscription = this.route.params
+      .subscribe(
+        (params: Params) => {
+          this.id = Number(params['id']);
+        }
+      );
+    this.createFamilyMemberForm();
+  }
+
+  /**
+   * Creates the add family member form.
```

Comment:
```suggestion
   * Creates the family member form.
```
Make sure you add all necessary comments in this component file. Take reference from the accounting section.